core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -126,7 +127,7 @@ final boolean needsToDisconnectWhenFinished() {
         return disconnectWhenFinished;
     }
 
-    static final class HttpResponseWrapper implements StreamWriter<HttpObject>, Runnable {
+    static final class HttpResponseWrapper implements StreamWriter<HttpObject>, TimeoutController {
```

Comment:
Ditto - Do you think it's a good idea to make this class extend `DefaultTimeoutController`? Would it still hold when we add support for other timeouts, like intra-stream idle timeout?